### PR TITLE
Add bluepyefe and igor packages

### PR DESCRIPTION
**Pull request by DrTaDa**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-bluepyefe/package.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+
+class PyBluepyefe(PythonPackage):
+    """ Blue Brain Python E-feature extraction"""
+
+    homepage = "https://github.com/BlueBrain/BluePyEfe"
+    url = "https://pypi.io/packages/source/b/bluepyefe/bluepyefe-0.3.13.tar.gz"
+
+    version('develop', branch='master')
+    version('0.3.13', commit='163ee6e48c4091c0a2347800d85c61d357cf26b4')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-bluepyefe/package.py]: Tags would be preferred :)

File: var/spack/repos/builtin/packages/py-bluepyefe/package.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack import *
+
+
+class PyBluepyefe(PythonPackage):
+    """ Blue Brain Python E-feature extraction"""
+
+    homepage = "https://github.com/BlueBrain/BluePyEfe"
+    url = "https://pypi.io/packages/source/b/bluepyefe/bluepyefe-0.3.13.tar.gz"
+
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-bluepyefe/package.py]:
```suggestion

```

File: var/spack/repos/builtin/packages/py-bluepyefe/package.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    depends_on('py-scipy', type='run')
+    depends_on('py-matplotlib', type='run')
+    depends_on('py-sh', type='run')
+    depends_on('py-neo', type='run')
+    depends_on('py-igor', type='run')
+
+    def setup_run_environment(self, env):
+        env.set('NEURON_INIT_MPI', "0")
+        env.unset('PMI_RANK')
+
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-bluepyefe/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-igor/package.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    homepage = "http://blog.tremily.us/posts/igor/"
+    url = "https://pypi.io/packages/source/i/igor/igor-0.3.tar.gz"
+
+    version('0.3', branch='master')
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-numpy', type='run')
+    depends_on('py-nose', type='run')
+    depends_on('py-matplotlib', type='run')
+
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-igor/package.py]:
```suggestion
```

**Review by matz-e — CHANGES_REQUESTED** [review]

**Comment by matz-e** [comment]
Emacs should have some configuration options to purge spurious whitespace…

File: var/spack/repos/builtin/packages/py-bluepyefe/package.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack import *
+
+
+class PyBluepyefe(PythonPackage):
+    """ Blue Brain Python E-feature extraction"""
+
+    homepage = "https://github.com/BlueBrain/BluePyEfe"
+    url = "https://pypi.io/packages/source/b/bluepyefe/bluepyefe-0.3.13.tar.gz"
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-bluepyefe/package.py:13]:
```suggestion
    url = "https://pypi.io/packages/source/b/bluepyefe/bluepyefe-0.3.13.tar.gz"

```
Sorry, there should still be a newline.

File: var/spack/repos/builtin/packages/py-igor/package.py [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+from spack import *
+
+
+class PyIgor(PythonPackage):
+    """igor: interface for reading binary IGOR files."""
+
+    homepage = "http://blog.tremily.us/posts/igor/"
+    url = "https://pypi.io/packages/source/i/igor/igor-0.3.tar.gz"
+
+    version('0.3', url=url)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-igor/package.py]: Please use `sha256` here for the checksum of the tarball.

**Review by matz-e — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-bluepyefe/package.py (hunk 1 above)
Comment (DrTaDa) [reply var/spack/repos/builtin/packages/py-bluepyefe/package.py]: What are tags ?

**Comment by DrTaDa** [comment]
It seems there is an error with the setuptools for py-igor:
`error: option --single-version-externally-managed not recognized`

Did I make a mistake by specifying https://github.com/BlueBrain/spack/blob/9895229d316720eb374150aa89a1ba025b1c37ea/var/spack/repos/builtin/packages/py-igor/package.py#L17 ?

File: var/spack/repos/builtin/packages/py-igor/package.py [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+
+class PyIgor(PythonPackage):
+    """igor: interface for reading binary IGOR files."""
+
+    homepage = "http://blog.tremily.us/posts/igor/"
+    url = "https://pypi.io/packages/source/i/igor/igor-0.3.tar.gz"
+
+    version('0.3', sha256='b04ac38c68fb81cf3167a9981dc5a20379112d40268bb72c5a8514dc8051abba')
+
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-igor/package.py:16]:
```suggestion

    def patch(self):
        setupfile = FileFilter('setup.py')
        setupfile.filter('distutils.core', 'setuptools')

```

This is necromancy, that package seems so old… try this modification to use some *modern* Python stuff, maybe it'll help

Comment (matz-e) [reply var/spack/repos/builtin/packages/py-igor/package.py:16]: Ideally, this would actually be below the block of `depends_on`…

**Review by matz-e — CHANGES_REQUESTED** [review]

**Review by matz-e — APPROVED** [review]

**Comment by pramodk** [comment]
@DrTaDa : can be merged?

**Comment by DrTaDa** [comment]
> @DrTaDa : can be merged?

@pramodk Yes! Thank you

**Comment by wvangeit** [comment]
@pramodk can you also include this in the default install on HBP systems? Thanks.

**Comment by pramodk** [comment]
@wvangeit : sure! So py-bluepyefe should be installed on external system as a module?
cc: @jorblancoa

**Comment by wvangeit** [comment]
@pramodk , yep

**Pull request merged**